### ISSUE TITLE
helper.getNode() returns null

### ISSUE THREAD

**Issue by Luidog** [node-red/node-red-node-test-helper#24]
Hello,

I seem to be having trouble getting a node using helper.getNode(). The node always returns null. Here is the code.

`/* global describe beforeEach afterEach it */

var should = require("should");
const helper = require("node-red-node-test-helper");
const create = require("../src/nodes/create.js");

helper.init(require.resolve("node-red"));

describe("Create Record Node", function() {
  beforeEach(function(done) {
    helper.startServer(done);
  });

  afterEach(function(done) {
    helper.unload();
    helper.stopServer(done);
  });

  it("should be loaded", function(done) {
    const testFlows = [{ id: "n1", type: "inject" }];
    helper.load(create, testFlows, function() {
      var n1 = helper.getNode("n1");
      n1.should.have.property('type', 'inject');
      done();
    });
  });
});
`

I have read and reread the documentation and I can't seem to find what I am doing wrong. Any thoughts?

**Comment by Luidog** [comment]
Hey folks, 

So I figured out a portion of this issue.  It turns out that if your node requires a configuration node to function, the configuration node must be passed in as well.

Here is an example:

`
const create = require("../src/nodes/create.js");
const config = require("../src/client/client.js");
helper.init(require.resolve("node-red"));

describe("Create Record Node", function() {
beforeEach(function(done) {
helper.startServer(done);
});

afterEach(function(done) {
helper.unload();
helper.stopServer(done);
});

it("should be loaded", function(done) {
const testFlows = [{ id: "n1", type: "inject" }];
helper.load([config,create], testFlows, function() {
var n1 = helper.getNode("n1");
n1.should.have.property('type', 'inject');
done();
});
});
});
`

I still can not successfully test any of my nodes that require a configuration node as I can't get the configuration node to accept the configuration in specified in the test flow. Any help here would be greatly appreciated.

**Comment by Luidog** [comment]
So I have read through this projects index file and found no clues to my issue. This issue is seems to be that the configuration node is never updated with the information my other nodes require. Specifically it seems that the client created in the configuration node here: 

https://github.com/Luidog/node-red-contrib-filemaker/blob/39aa7bf9ab56afad28834f585eaa885d7bebddeb/src/client/client.js#L12

Is never made available to any nodes that depend upon the configuration node. You can see this here:

https://github.com/Luidog/node-red-contrib-filemaker/blob/39aa7bf9ab56afad28834f585eaa885d7bebddeb/src/nodes/create.js#L13

This test illustrates the issue:

https://github.com/Luidog/node-red-contrib-filemaker/blob/39aa7bf9ab56afad28834f585eaa885d7bebddeb/test/create_spec.js#L27-L56

I really like node-red and I would very much like to continue to develop my project, but not being able to test nodes that require a configuration node is a significant blocker to any serious development. 

I apologize in advance if I am missing something straightforward. Please - any help is appreciated.

**Comment by Luidog** [comment]
I think I might have figured this out. It seems that the flow was not waiting on my configuration node to resolve its' promise. I made the non configuration node's input event an async function and I was able to successfully test my non configuration node. 

I am not sure if this is correct behavior for the node-red-test-helper as I have not run into any issues with this in node-red itself. I will close this issue and check on slack before opening up a new issue for this behavior.  

I found quite a few things that could be documented a bit better (like passing multiple nodes to test a node requiring configuration). If I get confirmation from project contributors that my approach to testing is appropriate I will submit a PR with documentation updates.
